src/main/java/org/opensearch/knn/plugin/KNNPlugin.java

```diff
@@ -159,7 +163,8 @@
         ModelCache.initialize(ModelDao.OpenSearchKNNModelDao.getInstance(), clusterService);
         KNNCircuitBreaker.getInstance().initialize(threadPool, clusterService, client);
         knnStats = new KNNStats(KNNStatsConfig.KNN_STATS);
-        return ImmutableList.of(knnStats);
+        modelDao = ModelDao.OpenSearchKNNModelDao.getInstance();
```

Comment:
One reason I do not want to do this is that it will call the constructor in OpenSearchKNNModelDao that creates cluster settings updaters. For users that dont use k-NN, I rather we leave as small of a footprint as possible. What do you think?

Reply:
Ack.